### PR TITLE
EES-3230 add edit data block links on release content tab

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Adds an 'Edit data block' link to data blocks on the edit release content tab.

### CODE REVIEW

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    render(
+      <EditableBlockWrapper onEdit={handleEdit}>
+        <div>Child block</div>
+      </EditableBlockWrapper>,
+    );
+
+    expect(
+      screen.getByRole('button', { name: 'Edit block' }),
+    ).toBeInTheDocument();
+    userEvent.click(screen.getByRole('button', { name: 'Edit block' }));
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx:29]: Could simplify this down to just the `userEvent.click`

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  });
+
+  test('renders the edit block button and calls `onEdit` when clicked when `onEdit` is provided', () => {
+    const handleEdit = jest.fn();
+    render(
+      <EditableBlockWrapper onEdit={handleEdit}>
+        <div>Child block</div>
+      </EditableBlockWrapper>,
+    );
+
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx:25]: Best to assert that the `handleEdit` hasn't been called yet before doing pressing the button to prove that the button triggered it.

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    userEvent.click(screen.getByRole('button', { name: 'Remove block' }));
+
+    const modal = within(screen.getByRole('dialog'));
+
+    await waitFor(() => {
+      expect(modal.getByText('Remove block')).toBeInTheDocument();
+    });
+
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx:98]: Best to assert that `handleDelete` hasn't been called yet to prove that it's the confirmation that triggers it.

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx (hunk 1 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx]: yep done.

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx (hunk 2 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx:25]: Done

File: src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx (hunk 3 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/components/editable/__tests__/EditableBlockWrapper.test.tsx:96]: Done

**Review by ntsim — APPROVED** [review]

**Pull request merged**